acquisition.py

```diff
@@ -86,7 +91,87 @@ def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
         inputs = pad_sequence(inputs_af, batch_first=True, padding_value=0.0)
 
         self.load_best_proxy()
-        self.proxy.model.eval()
+        self.proxy.model.train()
         with torch.no_grad():
             outputs = self.proxy.model(inputs, input_afs_lens, None)
         return outputs
+
+
+class AcquisitionFunctionUCB(AcquisitionFunctionBase):
+    def __init__(self, config, proxy):
+        super().__init__(config, proxy)
+
+    def load_best_proxy(self):
+        super().load_best_proxy()
+
+    def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
+        super().get_reward_batch(inputs_af_base)
+
+        inputs_af = list(map(torch.tensor, inputs_af_base))
+        input_afs_lens = list(map(len, inputs_af_base))
+        inputs = pad_sequence(inputs_af, batch_first=True, padding_value=0.0)
+
+        self.load_best_proxy()
+        self.proxy.model.train()
+        with torch.no_grad():
+            outputs = (
+                torch.hstack([self.proxy.model(inputs, input_afs_lens, None)])
+                .cpu()
+                .detach()
+                .numpy()
+            )
+
+        mean = np.mean(outputs, axis=1)
+        std = np.std(outputs, axis=1)
+        score = mean + self.config.acquisition.ucb.kappa * std
+        score = torch.Tensor(score)
+        score = score.unsqueeze(1)
+        return score
+
+
+class AcquisitionFunctionEI(AcquisitionFunctionBase):
+    def __init__(self, config, proxy):
+        super().__init__(config, proxy)
+
+    def load_best_proxy(self):
+        super().load_best_proxy()
+
+    def getMinF(self, inputs, input_len, mask):
+        # inputs = torch.Tensor(inputs).to(self.config.device)
+
+        if self.config.proxy.model.lower() == "mlp":
+            outputs = self.proxy.model(inputs).cpu().detach().numpy()
```

Comment:
Is there a way not be forced to distinguish different ways of calling the models : self.model(inputs, **kwargs) for all proxy models ?  That would remove the need to distinguish between different proxies for all acquisition functions

Reply:
Yeah yeah, you made a similar commenta few days ago. I forgot to update this branch when the PR for a uniform model was merged because I am not really working on this branch. But yes, as you suggest there is one way to call all proxy models. I will update this branch